client/js/controllers/editor.js

```diff
@@ -831,6 +831,64 @@
                     }
                 };
 
+             $scope.importFile = function () {
+                    var file = filePath;
+                    if (file) {
+                        if (isLeaf) {
+                            var cached = file.split("/");
+                            cached.pop();
+                            file = cached.join("/");
+                        }
+                        var dialog = $('<div></div>').
+                                   html($compile('<form ng-controller="editor"> <input' +
```

Comment:
AFAIK, AngularJS does not allow declaration of the the same controller in two different forms. 

Also, the $compile and $scope would inject the new code inside the editor controller so this is form is not needed.